### ISSUE TITLE
"Ambiguous containers hierarchy" when using nested Providers.

### ISSUE THREAD

**Issue by vetaltor** [Kukkimonsuta/inversify-react#18]
Hi, 
I'm trying to use multiple nested Providers, each with it's own container: top level container with services shared across app (logging, http, etc.) and nested container with module specific services. 

But i'm getting:

> Error: Ambiguous containers hierarchy.
> Provider has found a parent for specified `container`, but it already has a parent.

Although it's mentioned in [docs](https://github.com/Kukkimonsuta/inversify-react?tab=readme-ov-file#provider) that such case is valid.

> can automatically establish [hierarchy of containers](https://github.com/inversify/InversifyJS/blob/master/wiki/hierarchical_di.md) in React tree when you use multiple Providers (e.g. in a big modular app)

**Comment by vetaltor** [comment]
Here is the simplified example.

```tsx
import { Provider, useInjection } from 'inversify-react';
import { Container } from 'inversify';

const globalContainer = new Container({
  defaultScope: 'Singleton',
});
globalContainer.bind('globalService').toConstantValue('globalService');

const moduleContainer = new Container({
  defaultScope: 'Singleton',
});
moduleContainer.bind('moduleService').toConstantValue('moduleService');

function App() {
  return (
    <Provider container={globalContainer} key={globalContainer.id} >
      <Provider container={moduleContainer} key={moduleContainer.id} >
        <Module/>
      </Provider>
    </Provider>
  );
}

function Module() {
  const x = useInjection<string>('moduleService');
  return x;
}

**Comment by vetaltor** [comment]
I found out this error happens when wrapping in React's strict mode.
```tsx
<React.StrictMode>
  <App />  
</React.StrictMode>

**Comment by Kukkimonsuta** [comment]
Sorry for late response, this issue should be fixed. `React.StrictMode` renders things twice which triggered hierarchy check which didn't account for option that containers are already setup in correct hierarchy.

Fix was published in version 1.1.1 https://www.npmjs.com/package/inversify-react
